src/libcollections/btree/map.rs

```diff
@@ -875,20 +874,30 @@ impl<K: Ord, V: Ord> Ord for BTreeMap<K, V> {
     }
 }
 
-#[stable]
-impl<K: Show, V: Show> Show for BTreeMap<K, V> {
-    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        try!(write!(f, "BTreeMap {{"));
-
-        for (i, (k, v)) in self.iter().enumerate() {
-            if i != 0 { try!(write!(f, ", ")); }
-            try!(write!(f, "{:?}: {:?}", *k, *v));
-        }
+macro_rules! fmt_btree_map {
```

Comment:
Perhaps these macros could be refactored to generic functions? I think basically all of them work off some form of iterator, so perhaps that could be a private implementation detail of this library for now?